share/shwap/p2p/bitswap/getter.go

```diff
@@ -0,0 +1,172 @@
+package bitswap
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ipfs/boxo/blockstore"
+	"github.com/ipfs/boxo/exchange"
+
+	"github.com/celestiaorg/celestia-app/pkg/wrapper"
+	"github.com/celestiaorg/rsmt2d"
+
+	"github.com/celestiaorg/celestia-node/header"
+	"github.com/celestiaorg/celestia-node/share"
+)
+
+// Getter implements share.Getter.
+type Getter struct {
+	exchange exchange.SessionExchange
+	bstore   blockstore.Blockstore
+	session  exchange.Fetcher
+	cancel   context.CancelFunc
+}
+
+// NewGetter constructs a new Getter.
+func NewGetter(exchange exchange.SessionExchange, bstore blockstore.Blockstore) *Getter {
+	return &Getter{exchange: exchange, bstore: bstore}
+}
+
+// Start kicks off internal fetching session.
+func (g *Getter) Start() {
+	ctx, cancel := context.WithCancel(context.Background())
+	g.session = g.exchange.NewSession(ctx)
+	g.cancel = cancel
+}
+
+// Stop shuts down Getter's internal fetching session.
+func (g *Getter) Stop() {
+	g.cancel()
+}
+
+// TODO(@Wondertan): Rework API to get coordinates as a single param to make it ergonomic.
+func (g *Getter) GetShares(
+	ctx context.Context,
+	hdr *header.ExtendedHeader,
+	rowIdxs, colIdxs []int,
+) ([]share.Share, error) {
+	if len(rowIdxs) != len(colIdxs) {
+		return nil, fmt.Errorf("row indecies and col indecies must be same length")
+	}
+
+	if len(rowIdxs) == 0 {
+		return nil, fmt.Errorf("empty coordinates")
+	}
+
+	blks := make([]Block, 0, len(rowIdxs))
+	for i, rowIdx := range rowIdxs {
+		sid, err := NewEmptySampleBlock(hdr.Height(), rowIdx, colIdxs[i], len(hdr.DAH.RowRoots))
+		if err != nil {
+			return nil, err
+		}
+
+		blks[i] = sid
```

Comment:
Should be append. Otherwise will panic on write to `blks` slice of len 0.

Reply:
"This PR does not intend to implement any unit tests." 💀